imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -0,0 +1,212 @@
+"""Decommutate HIT CCSDS data and create L1a data products."""
+
+import logging
+import typing
+from collections import defaultdict
+from dataclasses import fields
+from enum import IntEnum
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+
+from imap_processing import decom, imap_module_directory, utils
+from imap_processing.cdf.global_attrs import ConstantCoordinates
+from imap_processing.cdf.utils import write_cdf
+from imap_processing.hit import hit_cdf_attrs
+from imap_processing.hit.l0.data_classes.housekeeping import Housekeeping
+
+logger = logging.getLogger(__name__)
+
+
+class HitAPID(IntEnum):
+    """
+    HIT APID Mappings.
+
+    Attributes
+    ----------
+    HIT_HSKP: int
+        Housekeeping
+    HIT_SCIENCE : int
+        Science
+    HIT_IALRT : int
+        I-ALiRT
+    """
+
+    HIT_HSKP = 1251  # Housekeeping
+    HIT_SCIENCE = 1252  # Science
+    HIT_IALRT = 1253  # I-ALiRT
+
+
+def hit_l1a_data(packet_file: typing.Union[Path, str]):
+    """
+    Process HIT L0 data into L1A data products.
+
+    Parameters
+    ----------
+    packet_file : str
+        Path to the CCSDS data packet file.
+
+    Returns
+    -------
+    dict
+        List of file paths to CDF data product files
+    """
+    # Decom, sort, and group packets by apid
+    packets = decom_packets(packet_file)
+    sorted_packets = utils.sort_by_time(packets, "SHCOARSE")
+    grouped_data = group_data(sorted_packets)
+
+    # Create datasets
+    # TODO define keys to skip for each apid. Currently just have
+    #  a list for housekeeping. Some of these may change later.
+    #  leak_i_raw can be handled in the housekeeping class as an
+    #  InitVar so that it doesn't show up when you extract the object's
+    #  field names.
+    skip_keys = [
+        "shcoarse",
+        "ground_sw_version",
+        "packet_file_name",
+        "ccsds_header",
+        "leak_i_raw",
+    ]
+    datasets = create_datasets(grouped_data, skip_keys)
+
+    # Create CDF files
+    logger.info("Creating CDF files for HIT L1A data")
+    cdf_filepaths = []
+    for dataset in datasets.values():
+        cdf_file = write_cdf(dataset)
+        cdf_filepaths.append(cdf_file)
+    logger.info("L1A CDF files created")
```

Comment:
This function is great. My only note would be to add "cdf_filepaths" to the output logging here.